fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/properties/FDBRecordContextProperties.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * FDBRecordContextProperties.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.properties;
+
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.logging.LogMessageKeys;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FDBRecordContextProperties {
+    private static final FDBRecordContextProperties EMPTY_PROPERTIES = new FDBRecordContextProperties(Collections.emptyMap());
+
+    @Nonnull
+    private final Map<String, FDBPropertyKey<?>> properties;
+
+    private FDBRecordContextProperties(@Nonnull Map<String, FDBPropertyKey<?>> properties) {
+        this.properties = properties;
+    }
+
+    @Nonnull
+    public Map<String, FDBPropertyKey<?>> getProperties() {
+        return properties;
+    }
+
+    @Nonnull
+    public <T> T getProperty(@Nonnull FDBPropertyKey.Builder<T> propBuilder) {
```

Comment:
It feels weird to me that this takes a `.Builder` instead of just an `FDBPropertyKey`. In general, it would seem to me like `Builder`s should only be used to build `FDBPropertyKey`s, and then only `FDBPropertyKey`s should be used elsewhere

Reply:
Re-implemented the ```FDBPropertyKey``` and ```FDBRecordContextProperties``` so it uses the key instead of builder to load the value now.